### PR TITLE
Add example for reportWhen usage

### PR DESCRIPTION
**Pull request by jonkeller**
reportWhen, with documentation and unit tests, was added in #18847. This PR additionally adds an example.

### CODE REVIEW

File: examples/analytics-reportWhen.amp.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  </script>
+</head>
+<body>
+<h1 id="top">AMP Analytics reportWhen</h1>
+<div>
+  An event will be triggered once the image below becomes visible. But it won't be reported until the document exits
+  (e.g. an unload or pagehide event is fired). Look in the developer console, and then scroll to the bottom of the page
+  to show the image, and finally scroll back up and click the button just below this text. The network request will not
+  appear until the button is clicked.
```
Comment (jonkeller) [inline examples/analytics-reportWhen.amp.html]: Aside from the above paragraph, and of course the addition of reportWhen in the amp-analytics config, this file is largely the same as the existing analytics.amp.html, with large sections removed.

File: examples/analytics-reportWhen.amp.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+<span id="test2" class="box" data-vars-title="Mock the unload event" onclick="mockUnload();">
+  Click here to generate an unload event
+</span>
+<div>
+<amp-analytics id="analytics1">
+<script type="application/json">
+{
+  "transport": {"beacon": true, "xhrpost": false},
+  "requests": {
+    "endpoint": "/analytics"
```
Comment (lannka) [inline examples/analytics-reportWhen.amp.html]: it will be helpful to include some time elapse metrics here

Comment (jonkeller) [reply examples/analytics-reportWhen.amp.html]: Agreed...though this is not a real endpoint anyway.

Comment (jonkeller) [reply examples/analytics-reportWhen.amp.html]: Done

**Pull request merged**